cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1439,23 +1439,34 @@ type compileBicepResult struct {
 }
 
 func (p *BicepProvider) compileBicep(
-	ctx context.Context, modulePath string,
+	ctx context.Context,
+	modulePath string,
+) (*compileBicepResult, error) {
+	currentPrincipalId, err := p.curPrincipal.CurrentPrincipalId(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("fetching current principal id for bicepparam compilation: %w", err)
+	}
+	return compileBicep(ctx, modulePath, p.env, currentPrincipalId, p.bicepCli)
+}
+
+func compileBicep(
```

Comment:
Just wondering - I think you extracted so you could test it directly with your mock implementation of the CLI.  I have a slight preference for us to just keep the old design (where we just have a single, `compileBicep` that is an instance method of `*BicepProvider` and then in the test when we create the BicepProvider pass in the Mock CLI (it looked like the NewBicepProvider method would allow us to do that) or failing that, downcast the return type of NewBicepProvider into a `*BicepProvider` and wack the private bicepCli member with our mock.  That should be possible because the test is in the same package as the implementation and can see private member state.  I think that would be a little more idiomatic and would match how we do this in some other places.

Could that be made to work?

Reply:
> Could that be made to work?

I think so. I will do/try it. 
Thank you for the suggestion; it is a good one and definitely better

Reply:
updated.